lms/djangoapps/grades/models.py

```diff
@@ -236,9 +241,43 @@ def bulk_get_or_create(cls, block_record_lists, course_key):
         BlockRecordList objects for the given course_key, but
         only for those that aren't already created.
         """
-        existent_records = {record.hashed: record for record in cls.bulk_read(course_key)}
+        existent_records = cls.bulk_read(course_key)
         non_existent_brls = {brl for brl in block_record_lists if brl.hash_value not in existent_records}
-        cls.bulk_create(non_existent_brls)
+        created = cls.bulk_create(non_existent_brls)
+        cls.update_cache(course_key, created)
+
+
+    @classmethod
+    def prefetch(cls, course_key):
```

Comment:
Can you update the names of the internal-only methods to start with underscore?  It will help with readability so we understand which ones are implementation-details.